### DISCUSSION TITLE
Creating a custom route builder with AOP but without interceptor beans in Kotlin

### DISCUSSION THREAD

**Discussion by ThraaxSession** [micronaut-projects/micronaut-core#7796] [Q&A] [answered]
Hi together,

I want to write a custom route  builder, but I can't find any documentations how to implement it.
My service holds some information like HTTP Method and the custom endpoint. Each time this custom endpoint gets a request, this service will execute depending on the data it's receives from the endpoint.

I've asked this already on stackoverflow, but it didn't help as the API might already changed.
[https://stackoverflow.com/questions/69813651/micronaut-core-how-to-create-dynamic-endpoints](https://stackoverflow.com/questions/69813651/micronaut-core-how-to-create-dynamic-endpoints)

Can someone help with it? A MethodInterceptor won't work as I don't have a class where I can annotate it. I only have custom informations in the database, which will be created by the user.

**Comment by yawkat** [comment, marked as answer]
maybe you could use a filter?

**Reply by ThraaxSession** [reply]
Hi @yawkat oh that might be a good idea. And I can even filter the uri! :) I'll try it and mark your answer then.

**Reply by ThraaxSession** [reply]
Sorry @yawkat I've got another question to the filter. Reactor ist pretty new to me. How would i publish the `MutableHttpResponse` without executing the further chain?
![image](https://user-images.githubusercontent.com/3085830/182134821-286cef57-906f-4bea-b860-596da9c6c609.png)

**Reply by ThraaxSession** [reply]
Ah got it. Was easier than I thought. :D
![image](https://user-images.githubusercontent.com/3085830/182136284-8ceb48e9-1940-4388-a188-0eb8c1904781.png)
